L1Trigger/Phase2L1ParticleFlow/interface/egamma/pftkegsorter_barrel_ref.h

```diff
@@ -0,0 +1,202 @@
+#ifndef REF_PFTKEGSORTER_BARREL_REF_H
+#define REF_PFTKEGSORTER_BARREL_REF_H
+
+#include <cstdio>
+#include <vector>
+
+#include "DataFormats/L1TParticleFlow/interface/layer1_emulator.h"
+#include "L1Trigger/Phase2L1ParticleFlow/interface/common/bitonic_hybrid_sort_ref.h"
+
+#ifdef CMSSW_GIT_HASH
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#endif
+
+namespace edm {
+  class ParameterSet;
+}
+
+namespace l1ct {
+  class PFTkEGSorterBarrelEmulator {
+  public:
+    PFTkEGSorterBarrelEmulator(const unsigned int nObjToSort = 10, const unsigned int nObjSorted = 16)
+        : nObjToSort_(nObjToSort), nObjSorted_(nObjSorted), debug_(false) {}
+
+    PFTkEGSorterBarrelEmulator(const edm::ParameterSet& iConfig);
+
+    virtual ~PFTkEGSorterBarrelEmulator() {}
+
+    void setDebug(bool debug = true) { debug_ = debug; };
+
+    void toFirmware_pho(const OutputRegion& outregions, EGIsoObj (&photons_in)[nObjSorted_]) const {
+      for (unsigned int i = 0; i < nObjToSort_; i++) {
+        EGIsoObj pho;
+        if (i < outregions.egphoton.size()) {
+          pho = outregions.egphoton[i];
+        } else
+          pho.clear();
+
+        photons_in[i] = pho;
+      }
+    }
+
+    void toFirmware_ele(const OutputRegion& outregions, EGIsoEleObj (&eles_in)[nObjSorted_]) const {
+      for (unsigned int i = 0; i < nObjToSort_; i++) {
+        EGIsoEleObj ele;
+        if (i < outregions.egelectron.size()) {
+          ele = outregions.egelectron[i];
+        } else
+          ele.clear();
+
+        eles_in[i] = ele;
+      }
+    }
+
+    template <typename T>
+    void run(const std::vector<PFInputRegion>& pfregions,
+             const std::vector<OutputRegion>& outregions,
+             const std::vector<unsigned int>& region_index,
+             std::vector<T>& eg_sorted_inBoard) {
+      // we copy to be able to resize them
+      std::vector<std::vector<T>> objs_in;
+      objs_in.reserve(nObjToSort_);
+      for (unsigned int i : region_index) {
+        std::vector<T> objs;
+        extractEGObjEmu(pfregions[i].region, outregions[i], objs);
+        if (debug_)
+          std::cout << "objs size " << objs.size() << "\n";
```

Comment:
Replace the `std::cout` with `dbgCout()` from https://github.com/p2l1pfp/cmssw/blob/L1PF_12_3_X/L1Trigger/Phase2L1ParticleFlow/interface/dbgPrintf.h